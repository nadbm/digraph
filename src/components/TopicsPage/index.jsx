// @flow
import React from 'react'
import { graphql, createFragmentContainer } from 'react-relay'
import { isEmpty } from 'ramda'

import List from '../ui/List'
import Topic from '../ui/Topic'
import { liftNodes } from '../../utils'

type Props = {
  organization: {
    topics: Object,
  },
}

const TopicsPage = ({ organization, ...props }: Props) => {
  const topics = liftNodes(organization.topics)
  return (
    <div>
      <div className="Subhead">
        <div className="Subhead-heading">Topics</div>
      </div>
      <List
        placeholder="There are no topics"
        hasItems={!isEmpty(topics)}
      >
        { topics.map(topic => (
          <Topic
            key={topic.uid}
            topic={topic}
            organization={organization}
            {...props}
          />
        )) }
      </List>
    </div>
  )
}

export const query = graphql`
query TopicsPage_query_Query($organizationId: String!) {
  organization(externalId: $organizationId) {
    ...TopicsPage_organization
  }
}`

export default createFragmentContainer(TopicsPage, graphql`
  fragment TopicsPage_organization on Organization {
    id
    ...Topic_organization

    topics(first: 1000) @connection(key: "Organization_topics") {
      edges {
        node {
          uid
          ...Topic_topic
        }
      }
    }
  }
`)
