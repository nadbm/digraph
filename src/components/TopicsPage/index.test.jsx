import React from 'react'
import { shallow } from 'enzyme'
import TopicsPage from './index'

jest.mock('react-relay', () => ({ createFragmentContainer: component => component }))

describe('<TopicsPage />', () => {
  const viewer = {
    name: 'Rezrov',
  }

  const organization = {
    resourceId: 'organization:tyrell',
    topics: {
      edges: [
        {
          node: {
            name: 'Frotz',
            uid: '1234',
          },
        },
      ],
    },
  }

  const wrapper = shallow(
    <TopicsPage
      organization={organization}
      viewer={viewer}
    />,
  )

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
