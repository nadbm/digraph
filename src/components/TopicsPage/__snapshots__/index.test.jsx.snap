// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicsPage /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopicsPage
    organization={
      Object {
        "resourceId": "organization:tyrell",
        "topics": Object {
          "edges": Array [
            Object {
              "node": Object {
                "name": "Frotz",
                "uid": "1234",
              },
            },
          ],
        },
      }
    }
    viewer={
      Object {
        "name": "Rezrov",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Subhead"
        >
          <div
            className="Subhead-heading"
          >
            Topics
          </div>
        </div>,
        <Unknown
          hasItems={true}
          placeholder="There are no topics"
        >
          <Topic
            organization={
              Object {
                "resourceId": "organization:tyrell",
                "topics": Object {
                  "edges": Array [
                    Object {
                      "node": Object {
                        "name": "Frotz",
                        "uid": "1234",
                      },
                    },
                  ],
                },
              }
            }
            topic={
              Object {
                "name": "Frotz",
                "uid": "1234",
              }
            }
            viewer={
              Object {
                "name": "Rezrov",
              }
            }
          />
        </Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="Subhead-heading"
          >
            Topics
          </div>,
          "className": "Subhead",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Topics",
            "className": "Subhead-heading",
          },
          "ref": null,
          "rendered": "Topics",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Topic
              organization={
                Object {
                  "resourceId": "organization:tyrell",
                  "topics": Object {
                    "edges": Array [
                      Object {
                        "node": Object {
                          "name": "Frotz",
                          "uid": "1234",
                        },
                      },
                    ],
                  },
                }
              }
              topic={
                Object {
                  "name": "Frotz",
                  "uid": "1234",
                }
              }
              viewer={
                Object {
                  "name": "Rezrov",
                }
              }
            />,
          ],
          "hasItems": true,
          "placeholder": "There are no topics",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1234",
            "nodeType": "class",
            "props": Object {
              "organization": Object {
                "resourceId": "organization:tyrell",
                "topics": Object {
                  "edges": Array [
                    Object {
                      "node": Object {
                        "name": "Frotz",
                        "uid": "1234",
                      },
                    },
                  ],
                },
              },
              "topic": Object {
                "name": "Frotz",
                "uid": "1234",
              },
              "viewer": Object {
                "name": "Rezrov",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Subhead"
          >
            <div
              className="Subhead-heading"
            >
              Topics
            </div>
          </div>,
          <Unknown
            hasItems={true}
            placeholder="There are no topics"
          >
            <Topic
              organization={
                Object {
                  "resourceId": "organization:tyrell",
                  "topics": Object {
                    "edges": Array [
                      Object {
                        "node": Object {
                          "name": "Frotz",
                          "uid": "1234",
                        },
                      },
                    ],
                  },
                }
              }
              topic={
                Object {
                  "name": "Frotz",
                  "uid": "1234",
                }
              }
              viewer={
                Object {
                  "name": "Rezrov",
                }
              }
            />
          </Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="Subhead-heading"
            >
              Topics
            </div>,
            "className": "Subhead",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Topics",
              "className": "Subhead-heading",
            },
            "ref": null,
            "rendered": "Topics",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Topic
                organization={
                  Object {
                    "resourceId": "organization:tyrell",
                    "topics": Object {
                      "edges": Array [
                        Object {
                          "node": Object {
                            "name": "Frotz",
                            "uid": "1234",
                          },
                        },
                      ],
                    },
                  }
                }
                topic={
                  Object {
                    "name": "Frotz",
                    "uid": "1234",
                  }
                }
                viewer={
                  Object {
                    "name": "Rezrov",
                  }
                }
              />,
            ],
            "hasItems": true,
            "placeholder": "There are no topics",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1234",
              "nodeType": "class",
              "props": Object {
                "organization": Object {
                  "resourceId": "organization:tyrell",
                  "topics": Object {
                    "edges": Array [
                      Object {
                        "node": Object {
                          "name": "Frotz",
                          "uid": "1234",
                        },
                      },
                    ],
                  },
                },
                "topic": Object {
                  "name": "Frotz",
                  "uid": "1234",
                },
                "viewer": Object {
                  "name": "Rezrov",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
